Setting router ID for Juniper OSPF

haverma opened the pull request:
- Following the same flow as Cisco, but using min IPs instead of max
- Also making hostname of juniper_ospf_stub_areas consistent with the filename

arifogel commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> with one suggestion

---

Reviewed 8 of 8 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[projects/batfish/src/main/java/org/batfish/representation/juniper/JuniperConfiguration.java, line 2509 at r1](https://reviewable.io/reviews/batfish/batfish/1473#-LDDWXCd57Mp4ohorFvK:-LDDWXCeDcwby80yblzj:bp09zn0) ([raw file](https://github.com/batfish/batfish/blob/21acbc685915b6b6a1a773d785eb79831b1bcb8d/projects/batfish/src/main/java/org/batfish/representation/juniper/JuniperConfiguration.java#L2509)):*
> ```Java
>   }
> 
>   private Ip getRouterId(RoutingInstance routingInstance) {
> ```

Does this only apply to OSPF? If so, please change name of function to `getOspfRouterId` or something like that for clarity.

---


*Comments from [Reviewable](https://reviewable.io/reviews/batfish/batfish/1473#-:-LDDXU-s2p-J0gOHefdF:b3923ex)*
<!-- Sent from Reviewable.io -->

haverma commented:
changed name of helper

---

Review status: 5 of 8 files reviewed at latest revision, 1 unresolved discussion.

---

*[projects/batfish/src/main/java/org/batfish/representation/juniper/JuniperConfiguration.java, line 2509 at r1](https://reviewable.io/reviews/batfish/batfish/1473#-LDDWXCd57Mp4ohorFvK:-LDEd9IM4zMAKY28zX9k:b-n9tkhd) ([raw file](https://github.com/batfish/batfish/blob/21acbc685915b6b6a1a773d785eb79831b1bcb8d/projects/batfish/src/main/java/org/batfish/representation/juniper/JuniperConfiguration.java#L2509)):*
<details><summary><i>Previously, arifogel (Ari Fogel) wrote…</i></summary><blockquote>

Does this only apply to OSPF? If so, please change name of function to `getOspfRouterId` or something like that for clarity.
</blockquote></details>

changed

---


*Comments from [Reviewable](https://reviewable.io/reviews/batfish/batfish/1473#-:-LDEdAeQ22qo2lf-eKp6:b2w8qnn)*
<!-- Sent from Reviewable.io -->

arifogel commented:
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/batfish/batfish/1473)*
<!-- Sent from Reviewable.io -->
